### PR TITLE
This will add a singleheader script. 

### PR DESCRIPTION
**Pull request by lemire**
It builds a 'demo' file. Unfortunately, because we have a dependency on ICU (for example), one then needs to do extra work to link the result with the library.

I accidentally found out that we linked twice against ada when building our test executables.

### CODE REVIEW

File: singleheader/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+While in the ada main directory, using Python 3, type:
```
Comment (anonrig) [inline singleheader/README.md:1]: This file will be included in the documentation page, if we don't explicitly exclude it. If we want to add this, let's add a descriptive header using `#` prefix.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -17,7 +17,7 @@ link_libraries(ada)
 
 add_cpp_test(wpt_tests)
 target_link_libraries(wpt_tests PUBLIC simdjson)
-target_link_libraries(wpt_tests PUBLIC ada)
+# target_link_libraries(wpt_tests PUBLIC ada)
```
Comment (anonrig) [inline tests/CMakeLists.txt:20]: Can we revert these changes?

Comment (lemire) [reply tests/CMakeLists.txt]: Done.

File: singleheader/README.md (hunk 1 above)
Comment (lemire) [reply singleheader/README.md:3]: Right, but that should not be front-page. End-users should not ever have to do this.

**Review by anonrig — APPROVED** [review]

**Comment by lemire** [comment]
Merging.

**Comment by lemire** [comment]
@anonrig 

Note that the singleheader.zip can be added to a release package.

**Comment by lemire** [comment]
@lemire Building ada.cpp is easy, but the ICU dependency is... well... it can be trivial or very hard, depending on the system. So the final linking is tricky. Under Linux, it is trivial. Under macOS, it is requires a bit more care.

**Comment by anonrig** [comment]
@anonrig I'll add documentation for using and give an example of a Node.js Gyp integration for linking ICU with Ada.
